CHANGELOG.md

```diff
@@ -13,6 +13,8 @@ Compatibility:
 * Add `Exception#detailed_message` method (#3257, @andrykonchin).
 * Fix `rb_enc_vsprintf` and force String encoding instead of converting it (@andrykonchin).
 * Add `rb_gc_mark_movable` function (@andrykonchin).
+* Promoted `File#path` and `File#to_path` to `IO#path` and `IO#to_path` and made IO#new accept an optional `path:` keyword argument (#3275, @moste00)
```

Comment:
It's convenient to use imperative verbs in a commit title, so it could be "Promote ...".

Also there may be mentioned a related GitHub issue (if it exists), not a link to a pull request.

Reply:
Hey, I know I totally left this PR for ages, sorry !

This is done now, I squashed all commits and edited both the CHANGELOG and the commit message to use "promote". 

Seperating the deletion of the unused exception classes into a seperate commit is a bit hairy/tedious git manipulation, so I figured I can get lazy and squash everything. If it's absolutely necessary, let me know and I will do it at once.

Reply:
Looks great!

Could you please address two comments that are left? (changelog record wording and `options[:path]` hash lookup)

Reply:
Done now ! 

I have no idea how I tricked my brain into thinking that I already made the changes when I wrote the earlier comments haha.

Reply:
👍